src/postcondition.ts

```diff
@@ -93,11 +96,14 @@ export interface NonFungiblePostCondition {
 export function createNonFungiblePostCondition(
   principal: string | PostConditionPrincipal,
   conditionCode: NonFungibleConditionCode,
-  assetInfo: AssetInfo,
+  assetInfo: string | AssetInfo,
   assetName: ClarityValue
 ): NonFungiblePostCondition {
   if (typeof principal === 'string') {
-    principal = createStandardPrincipal(principal);
+    principal = parsePrincipalString(principal);
+  }
+  if (typeof assetInfo === 'string') {
+    assetInfo = parseAssetInfoString(assetInfo);
```

Comment:
Should `createNonFungiblePostCondition` be responsible for the type conditional here? 

Maybe a generic `parsePrincipal(arg: string | PostConditionPrincipal)` would reduce repetition?